packages/markdown/loader.js

```diff
@@ -142,6 +143,56 @@ module.exports = async function demoLoader() {
     ),
   );
 
+  /**
+   * @param {*} demoName
+   * @param {*} moduleFilepath
+   * @param {*} variant
+   * @param {*} importModuleID
+   * @example detectRelativeImports('ComboBox.js', '', JS', './top100Films') => relativeModules.set('ComboBox.js', new Map([['./top100Films.js', ['JS']]]))
+   */
+  function detectRelativeImports(demoName, moduleFilepath, variant, importModuleID) {
+    if (importModuleID.startsWith('.')) {
+      let relativeModuleFilename = importModuleID;
+      const demoMap = relativeModules.get(demoName);
+      // If the moduleID does not end with an extension, or ends with an unsupported extension (e.g. ".styling") we need to resolve it
+      // Fastest way to get a file extension, see: https://stackoverflow.com/a/12900504/
+      let importType = importModuleID.slice(((importModuleID.lastIndexOf('.') - 1) >>> 0) + 2);
+      const supportedTypes = ['js', 'jsx', 'ts', 'tsx', 'css', 'json'];
+      if (!importType || !supportedTypes.includes(importType)) {
+        // If the demo is a JS demo, we can assume that the relative import is either
+        // a `.js` or a `.jsx` file, with `.js` taking precedence over `.jsx`
+        // likewise for TS demos, with `.ts` taking precedence over `.tsx`
+        const extensions =
+          variant === 'JS' ? ['.js', '.jsx', '.ts', '.tsx'] : ['.ts', '.tsx', '.js', '.jsx'];
+        const extension = extensions.find((ext) => {
+          try {
+            return statSync(path.join(moduleFilepath, '..', `${importModuleID}${ext}`));
+          } catch (error) {
+            // If the file does not exist, we return false and continue to the next extension
+            return false;
+          }
+        });
+        if (!extension) {
+          throw new Error(
+            `You are trying to import a module "${importModuleID}" in the demo "${demoName}" that could not be resolved. Please make sure that one of: \n\t\t - ${extensions.map((ext) => `${importModuleID}${ext}`).join('\n\t\t - ')} \n exists.`,
+          );
```

Comment:
```suggestion
          throw new Error(
            [
              `You are trying to import a module "${importModuleID}" in the demo "${demoName}" that could not be resolved.`,
              `Please make sure that one of the following file exists:`,
              ...extensions.map((ext) => `- ${importModuleID}${ext}`),
            ].join('\n'),
          );
```

Here is a solution we frequently use in X to have long and readable error message. Instead of using the `\n` you can generate an array and join it with `\n` such that you visually see the braking lines.

Reply:
Excellent, thanks a ton! 😅